---
title: "Shotgun - Functional Annotation Results"
author: "Nate Olson"
date: "6/29/2018"
output: html_document
---

```{r}
library(ProjectTemplate)
load.project()
library(lubridate)
library(readxl)
```

calculating proportions
```{r}
gene_prop <- funct_anno_df %>% 
    group_by(Sample_ID) %>% 
    mutate(total_count = n()) %>% 
    group_by(Sample_ID, predicted_gene_name) %>% 
    mutate(gene_count = n(),
           gene_prop = gene_count / total_count) %>% 
    select(Sample_ID, predicted_gene_name, gene_count, total_count, gene_prop) %>% 
    distinct()
```


```{r}
ar_genes <- read_excel("data/raw_data/ar_tox_genes.xlsx", 
                       sheet = 1, range = "B2:C320") %>% 
    dplyr::rename(gene_name = toupper(`Relevant genes from ARDB`)) %>%
    mutate(predicted_gene_name = toupper(gene_name))

anno_gene_filt <- gene_prop %>% 
    left_join(ar_genes)
```

Cleaning and formatting metadata
```{r}
ar_gene_df <- anno_gene_filt %>% 
    separate(Sample_ID,
             c("Sample_Code", "Collection_Date", "Sample_Type"), 
             sep = "_") %>%   
    #Fix date format
    mutate(Collection_Date = mdy(paste0(Collection_Date,"-2017")))

## Comparing samples before and after zvi filteration
ar_rw_zw_df <- ar_gene_df %>% 
    filter(Sample_Type == "W", Sample_Code %in% c("RW","ZVI")) %>%
    #Change Sample_Codes to be consistent with 16S analysis
    mutate(Sample_Code = if_else(Sample_Code == "ZVI", "ZW", Sample_Code))
```


Number of antibiotic genes per sample
```{r}
## Need to revise for gene proportions
# anno_gene_filt_tbl <- anno_gene_filt %>% 
#     filter(!is.na(gene_name)) %>% 
#     group_by(Antibiotic, gene_name, Sample_ID) %>% 
#     summarise(count = n()) %>% 
#     spread(Sample_ID, count, fill = 0)
```




Looking at Chloramphenicol by sample as an example. 
Pick chloramphenicol few genes with a good number of observations, basically good for plotting.
```{r}
# Bar plot of all chloramphenicol gene differences between RW and ZW
ar_rw_zw_df %>% #filter(Antibiotic == "Chloramphenicol") %>% 
    filter(!is.na(gene_name)) %>% 
    ggplot() + 
    geom_point(aes(x = Collection_Date, y = gene_prop, fill = Sample_Code), 
             stat = "identity", shape = 21, color = "grey20") + 
    facet_wrap(~gene_name, scales = "free_x") + 
    labs(x = "Collection Date", fill = "Sample_Code") + 
    theme_bw() + 
    theme(axis.text.x = element_text(angle = -45, hjust = 0))
``` 


```{r}
# Include only CE and RB samples
ar_ce_rb_df <- ar_gene_df %>% 
    filter(Sample_Type == "W", Sample_Code %in% c("MAN","RBC")) %>%
    #Change Sample_Codes to be consistent with 16S analysis
    mutate(Sample_Code = if_else(Sample_Code == "MAN", "CE", Sample_Code),
           Sample_Code = if_else(Sample_Code == "RBC", "RB", Sample_Code))
  
## Bar plot of all chloramphenicol gene differences between RW and ZW
ar_ce_rb_df %>%  filter(Antibiotic == "Chloramphenicol") %>%
    ggplot() + 
    geom_bar(aes(x = as.factor(Collection_Date), 
                 fill = Sample_Code), 
             color = "grey20", position = "dodge", width = 0.35) + 
    facet_wrap(~gene_name, scales = "free_x", nrow = 1) + 
    labs(x = "Collection Date", fill = "Sample_Code") + 
    theme_bw() + theme(axis.text.x = element_text(angle = -45, hjust = 0))
```


## Creating an MRexperiment object with function data
```{r}
funct_anno_df
```

```{r}
wide_fun_counts <- funct_anno_df %>% 
    ## Sample/ gene counts
    group_by(Sample_ID, predicted_gene_name) %>% summarize(count = n()) %>% 
    ## Converting to matrix
    spread(Sample_ID, count, fill = 0) %>% 
    mutate(predicted_gene_name = if_else(is.na(predicted_gene_name), 
                                         "uncategorized", 
                                         predicted_gene_name))
fun_mat <- ungroup(wide_fun_counts) %>% 
    as.data.frame() %>% 
    column_to_rownames(var = "predicted_gene_name") %>% 
    as.matrix()
```

```{r}
pheno_df <- data_frame(Sample_ID = colnames(fun_mat)) %>% 
    separate(Sample_ID, 
             c("Sample_Code", "Collection_Date", "Sample_Type"), 
             sep = "_", remove = FALSE) %>%   
    #Fix date format
    mutate(Collection_Date = mdy(paste0(Collection_Date,"-2017")))
pd <- as.data.frame(pheno_df)
rownames(pd) <- pd$Sample_ID    
```


```{r}
fun_mrexp <- newMRexperiment(fun_mat, phenoData = AnnotatedDataFrame(pd))
```

```{r}
fun_mrexp
```


Note citation information from developer

CITATION:
If you use this software, please cite:

[1] Fast genome-wide functional annotation through orthology assignment by
      eggNOG-mapper. Jaime Huerta-Cepas, Kristoffer Forslund, Luis Pedro Coelho,
      Damian Szklarczyk, Lars Juhl Jensen, Christian von Mering and Peer Bork.
      Mol Biol Evol (2017). doi: https://doi.org/10.1093/molbev/msx148

[2] eggNOG 4.5: a hierarchical orthology framework with improved functional
      annotations for eukaryotic, prokaryotic and viral sequences. Jaime
      Huerta-Cepas, Damian Szklarczyk, Kristoffer Forslund, Helen Cook, Davide
      Heller, Mathias C. Walter, Thomas Rattei, Daniel R. Mende, Shinichi
      Sunagawa, Michael Kuhn, Lars Juhl Jensen, Christian von Mering, and Peer
      Bork. Nucl. Acids Res. (04 January 2016) 44 (D1): D286-D293. doi:
      https://doi.org/10.1093/nar/gkv1248

[3] Fast and Sensitive Protein Alignment using DIAMOND. Buchfink B, Xie C,
       Huson DH. 2015. Nat. Methods [Internet] 12.


(e.g. Functional annotation was performed using emapper-0.99.2-3-g41823b2 [1]
 based on eggNOG orthology data [2]. Sequence searches were performed
 using [3].)
