---
title: "ZVI 16S rRNA dataset characterization"
author: "Prachi Kulkarni"
date: '`r Sys.Date()`'
output:
  bookdown::html_document2: 
    toc: FALSE
---

```{r message=FALSE, warning = FALSE, echo = FALSE}
## Loading project
library(ProjectTemplate)
load.project()
# Load required packages
library(knitr)
library(metagenomeSeq)
library(ggExtra)
```

```{r}
count_dat <- metagenomeSeq::expSummary(mrexp)
count_dat$Seq_ID <- rownames(count_dat)
count_tbl <- mrexp@assayData$counts %>% as_tibble()
count_df <- as.data.frame(count_dat)
```

```{r}
count_tbl$OTU <- mrexp@featureData@data$OTU
count_tbl <- count_tbl %>% gather("sam","count",-OTU) %>% 
    mutate(count = if_else(count != 0, 1,0)) %>% 
    group_by(OTU) %>% summarise(n_sam = sum(count)) %>% 
    filter(n_sam == 1)

#### Getting error message here when when trying to find assigned and unassigned OTUs - "`by` required, because the data sources have no common variables"
##  This issue is with the left_join statement there are no shared column names
## The following code runs but does not identify unassigned OTUs
## Emmanuel used de novo clustering so it looks like all the features are unassigned
# assigned_un_df <- count_tbl %>% 
#     left_join(mrexp@featureData@data, by = c("OTU" = "OTUname")) %>% 
#     mutate(assinged_un = if_else(grepl("OTU_", OTU), "unassigned","assigned")) %>%
#     group_by(assinged_un) %>% summarise(count = n())
```

All features are assigned to Bacteria or Archaea.
```{r}
## Getting the number of OTUs per taxa
taxa_tbl <- fData(mrexp) %>% as.data.frame() 
taxa_tbl$Rank1 %>% unique()
```
Only 2 OTUs are not assigned to the phylum level
```{r}
taxa_tbl$Rank2 %>% table()
```

The two features that are not assigned to the phylumn level are low abundance. 
No need to worry about filtering them as they are unlikely to affect the results. 
```{r}
count_dat <- MRcounts(mrexp, norm = FALSE, log = FALSE, sl = 1)
count_sum_df <- data_frame(OTUname = rownames(count_dat), total_norm_abu = rowSums(count_dat))
taxa_tbl %>% left_join(count_sum_df) %>% filter(Rank2 == "p__")
```

```{r}
# total seq count
count_total <- sum(count_df$libSize)
# total number of samples
num_samples <- nrow(count_df)

# average seq count per sample
avg_seq_sample <- count_total/num_samples
avg_seq_sample

## Cant get the following answers due to above error 
# number of unique assigned species level OTUs
#n_unique_assigned_otus <- assigned_un_df$count[assigned_un_df$assinged_un == "assigned"]
# number of unique unassigned species level OTUs
#n_unique_unassigned_otus <- assigned_un_df$count[assigned_un_df$assinged_un == "unassigned"]
```

```{r}
# To Subset the mrexperiment object so as not to include background, soil, lettuce, negative control, ie only water samples
control_samples <- c("NI","BG","RL", "ZL", "TL", "RS", "ZS", "TS","NR","NC")
mrexp1 = mrexp[,!(pData(mrexp)$Sample_Code %in% control_samples)]
```

```{r}
# Coverage analysis with only water samples described using Sample_Code
div_est <- mrexp1  %>% MRcounts()  %>% DataInfo()
sample_dat <- pData(mrexp) %>% as.data.frame() %>% 
    rownames_to_column(var = "site")

div_est %>% left_join(sample_dat) %>% 
    ggplot(aes(x = n, y = SC)) +
  geom_point(aes(color =Sample_Code)) + 
  scale_x_log10() + 
  geom_vline(aes(xintercept = 100), linetype = 2, color = "grey 60") + 
  theme_bw() +
  labs(x = "Number of Sequences",y = "Coverage", color = "Sample Code") +
  theme(legend.position = "bottom")

#ggsave("Coverage_Plot_with_Sample_Code_April_2_2018.tiff", p2, height = 5, width = 5, dpi = 300)
```

```{r}
# Coverage analysis with all samples described using sample type
div_est <- mrexp  %>% MRcounts()  %>% DataInfo()
ggplot(div_est, aes(x = n, y = SC)) +
  geom_point(aes(color = mrexp$Sample_Type)) + scale_x_log10() + 
  geom_vline(aes(xintercept = 100), linetype = 2, color = "grey 60") + 
  theme_bw() +
  labs(x = "Number of Sequences",y = "Coverage") +
  theme(legend.position = "bottom")
    p3$labels$colour <- "Sample Type" # Change Legend Label
ggMarginal(p3, type = "histogram", margins = "x")
p4 = ggMarginal(p3, type = "histogram", margins = "x")
#ggsave("Coverage_Plot_with_Sample_Type_April_2_2018.tiff", p4, height = 5, width = 5, dpi = 300)
```

